### PR TITLE
[SPARK-43321][Connect][Followup] Better names for APIs used in Scala Client joinWith

### PR DESCRIPTION
**Pull request by zhenlineo**
### What changes were proposed in this pull request?
Directly use `struct` in the names to directly call out the underlying encoder is composed using a struct encoder.

### Why are the changes needed?
Addressing PR review feedbacks of https://github.com/apache/spark/pull/40997

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existing tests

### CODE REVIEW

**Comment by zhenlineo** [comment]
cc @hvanhovell

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/AgnosticEncoder.scala [hunk 1]
```diff
@@ -37,15 +37,14 @@ import org.apache.spark.util.Utils
...
- * An encoder is flattenable if it contains fields that can be fattened to a row.
  */
 trait AgnosticEncoder[T] extends Encoder[T] {
   def isPrimitive: Boolean
   def nullable: Boolean = !isPrimitive
   def dataType: DataType
   override def schema: StructType = StructType(StructField("value", dataType, nullable) :: Nil)
   def lenientSerialization: Boolean = false
-  def isFlattenable: Boolean = false
+  def isStruct: Boolean = false // set to true if it is a struct encoder
```
Comment (vicennial) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/AgnosticEncoder.scala:47]: Nit: comment seems a bit redundant

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/AgnosticEncoder.scala [hunk 2]
```diff
@@ -103,18 +102,18 @@ object AgnosticEncoders {
   }
 
   // Contains a sequence of fields. The fields can be flattened to columns in a row.
```
Comment (vicennial) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/AgnosticEncoder.scala:104]: Nit: can also remove the word "flattened" in this comment here I think

**Review by vicennial — APPROVED** [review]
LGTM

**Review by hvanhovell — APPROVED** [review]
LGTM

**Comment by zhenlineo** [comment]
cc @HyukjinKwon Can we merge this? Thanks.

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**